modules/perf/class.jetpack-perf-optimize-assets.php

```diff
@@ -0,0 +1,370 @@
+		return $this->defer_scripts && apply_filters( 'jetpack_perf_defer_script', $should_defer_script, $script->handle, $script->src );
+	}
+
+	private function should_remove_script( $script ) {
+		return $this->should_remove_asset( 'jetpack_perf_remove_script', $script );
+	}
+
+	private function should_inline_script( $script ) {
+		return ( $this->inline_scripts_and_styles || $this->inline_always ) && $this->should_inline_asset( 'jetpack_perf_inline_script', $script );
+	}
+
+	/** STYLES **/
+	public function filter_inline_styles( $src, $handle ) {
+		global $wp_scripts;
+
+		if ( is_admin() || ! isset( $wp_scripts->registered[$handle] ) ) {
+			return $src;
+		}
+
+		$style = $wp_scripts->registered[$handle];
+
+		if ( $this->should_inline_style( $style ) ) {
+			return '#';
+		}
+
+		return $src;
+	}
+
+	public function print_inline_styles( $tag, $handle, $href, $media ) {
+		global $wp_styles;
+
+		if ( is_admin() || ! isset( $wp_styles->registered[$handle] ) ) {
+			return $tag;
+		}
+
+		$style = $wp_styles->registered[$handle];
+
+		if ( $this->should_inline_style( $style ) ) {
+			return "<style type='text/css' media='$media'>" . file_get_contents( $style->extra['jetpack-inline-file'] ) . '</style>';
+		}
+
+		if ( $this->should_remove_style( $style ) ) {
+			return '';
+		}
+
+		return $tag;
+	}
+
+	private function should_inline_style( $style ) {
+		return ( $this->inline_scripts_and_styles || $this->inline_always ) && $this->should_inline_asset( 'jetpack_perf_inline_style', $style );
+	}
+
+	private function should_remove_style( $style ) {
+		return $this->should_remove_asset( 'jetpack_perf_remove_style', $style );
+	}
+
+	/** shared code **/
+
+	private function should_inline_asset( $filter, $dependency ) {
+		// inline anything local, with a src starting with /, or starting with site_url
+		$site_url = site_url();
+
+		$is_local_url = ( strncmp( $dependency->src, '/', 1 ) === 0 && strncmp( $dependency->src, '//', 2 ) !== 0 )
+			|| strpos( $dependency->src, $site_url ) === 0;
+
+		if ( $is_local_url && ! isset( $dependency->extra['jetpack-inline'] ) ) {
+			$dependency->extra['jetpack-inline'] = true;
+
+			$path = untrailingslashit( ABSPATH ) . str_replace( $site_url, '', $dependency->src );
+
+			if ( ! file_exists( $path ) ) {
+				$path = str_replace('/', DIRECTORY_SEPARATOR, str_replace( $site_url, '', $dependency->src ));
+
+				$prefix = explode( DIRECTORY_SEPARATOR, untrailingslashit( WP_CONTENT_DIR ) );
+				$prefix = array_slice( $prefix, 0, array_search( $path[1], $prefix ) - 1 );
+
+				$path = implode( DIRECTORY_SEPARATOR, $prefix ) . $path;
+			}
+
+			$dependency->extra['jetpack-inline-file'] = $path;
+		}
+
+		$should_inline = isset( $dependency->extra['jetpack-inline'] ) && $dependency->extra['jetpack-inline'];
+
+		$will_inline = apply_filters( $filter, $should_inline, $dependency->handle, $dependency->src ) && file_exists( $dependency->extra['jetpack-inline-file'] );
```

Comment:
`apply_filters( $filter`

That kind of syntax is going to be a problem for our parser to handle. Maybe we could do something like what Core does with dynamic values inside a filter? Here is an example:
https://developer.wordpress.org/reference/hooks/_wp_post_revision_field_field/